rocAL/source/readers/image/reader_factory.cpp

```diff
@@ -100,6 +101,12 @@ std::shared_ptr<Reader> create_reader(ReaderConfig config) {
             return ret;
         } break;
 #endif
+        case StorageType::NUMPY_DATA: {
+            auto ret = std::make_shared<NumpyDataReader>();
+            if (ret->initialize(config) != Reader::Status::OK)
+                throw std::runtime_error("NumpyDataReader cannot access the storage");
```

Comment:
I think rocAL library has special THROW function. Why is it not used here?

Reply:
The numpy reader instantiation is following the same template as the other reader instantiations in this file.